script/build.js

```diff
@@ -139,37 +136,14 @@ smith.use(rewriteVaDomains(BUILD_OPTIONS));
 // On the server, it can be accessed at BUILD_OPTIONS.buildSettings.
 // In the browser, it can be accessed at window.settings.
 smith.use(createBuildSettings(BUILD_OPTIONS));
+smith.use(checkBrokenLinks(BUILD_OPTIONS));
 
-if (BUILD_OPTIONS.watch) {
-  const watchPaths = {
-    [`${BUILD_OPTIONS.contentRoot}/**/*`]: '**/*.{md,html}',
-    [`${BUILD_OPTIONS.contentPagesRoot}/**/*`]: '**/*.{md,html}',
-  };
-  const watchMetalSmith = watch({ paths: watchPaths, livereload: true });
-  smith.use(watchMetalSmith);
-  smith.use(webpackMetalsmithConnect.watchAssets(BUILD_OPTIONS));
-} else {
-  smith.use(webpackMetalsmithConnect.compileAssets(BUILD_OPTIONS));
-
-  const isDevBuild = [environments.DEVELOPMENT, environments.VAGOVDEV].includes(
-    BUILD_OPTIONS.buildtype,
-  );
-  if (!isDevBuild) {
-    smith.use(addAssetHashes(BUILD_OPTIONS));
-  }
-
-  if (process.env.CHECK_BROKEN_LINKS !== 'no') {
-    smith.use(checkBrokenLinks(BUILD_OPTIONS));
-  }
-}
+configureAssets(smith, BUILD_OPTIONS);
 
 smith.use(createSitemaps(BUILD_OPTIONS));
-
-// Pages can contain an "alias" property in their metadata, which is processed into
-// separate pages that will each redirect to the original page.
 smith.use(createRedirects(BUILD_OPTIONS));
-
 smith.use(moveRemove(BUILD_OPTIONS));
+
 /* eslint-disable no-console */
 smith.build(err => {
   if (err) throw err;
```

Comment:
The logic for determining our course of action for assets was getting complicated, so I moved it into its own file.

Also, I moved the link checker above the asset compilation, because it was pretty annoying to wait for assets to finish compiling only to find a single broken link fail the build.